components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -1069,199 +916,200 @@ void BraveSyncServiceImpl::SendDeviceSyncRecord(const int &action,
+                                          int index) {
+  auto* node = parent->GetChild(index);
 
-  if (parent_id != -1) {
-    parent_folder_order = sync_obj_map_->GetOrderByLocalObjectId(
-      storage::ObjectMap::Type::Bookmark, std::to_string(parent_id));
-    DCHECK(!parent_folder_order.empty());
-  }
+  auto* prev_node = index == 0 ?
+    nullptr :
+    parent->GetChild(index - 1);
+  auto* next_node = index >= parent->child_count() ?
+    nullptr :
+    parent->GetChild(index + 1);
 
-  LOG(ERROR) << "TAGAB prev_item_order="<<prev_item_order;
-  LOG(ERROR) << "TAGAB next_item_order="<<next_item_order;
-  LOG(ERROR) << "TAGAB parent_folder_order="<<parent_folder_order;
+  std::string prev_node_order;
+  if (prev_node)
+    prev_node->GetMetaInfo("order", &prev_node_order);
 
-  content::BrowserThread::GetTaskRunnerForThread(content::BrowserThread::UI)
-    ->PostTask(FROM_HERE,
-               base::Bind(&BraveSyncServiceImpl::BookmarkAddedQueryNewOrderUiWork,
-                          base::Unretained(this), node_id, prev_item_order,
-                          next_item_order, parent_folder_order));
-}
-
-void BraveSyncServiceImpl::BookmarkAddedQueryNewOrderUiWork(
-  const int64_t &node_id,
-  const std::string &prev_item_order,
-  const std::string &next_item_order,
-  const std::string &parent_folder_order) {
-
-  LOG(ERROR) << "TAGAB brave_sync::BraveSyncServiceImpl::BookmarkAddedQueryNewOrderUiWork";
-  LOG(ERROR) << "TAGAB node_id="<<node_id;
-  LOG(ERROR) << "TAGAB prev_item_order="<<prev_item_order;
-  LOG(ERROR) << "TAGAB next_item_order="<<next_item_order;
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-  DCHECK(sync_client_);
+  std::string next_node_order;
+  if (next_node)
+    next_node->GetMetaInfo("order", &next_node_order);
 
-  DCHECK(!prev_item_order.empty() || !next_item_order.empty() || !parent_folder_order.empty());
+  PushRRContext(
+      prev_node_order, next_node_order, node->id(), jslib_const::kActionCreate);
+  sync_client_->SendGetBookmarkOrder(prev_node_order, next_node_order);
+  // responds in OnSaveBookmarkOrder
+}
 
-  if (prev_item_order.empty() && next_item_order.empty()) {
-    // Special case, both prev_item_order and next_item_order are empty
-    // Can happened when sync is initialized and add bookmark into empty folder
-    const std::string &order = parent_folder_order + ".1";
-    OnSaveBookmarkOrderInternal(order, node_id, jslib_const::kActionCreate);
-  } else {
-    PushRRContext(prev_item_order, next_item_order, node_id, jslib_const::kActionCreate);
-    sync_client_->SendGetBookmarkOrder(prev_item_order, next_item_order);
-    // See later in OnSaveBookmarkOrder
-  }
+void BraveSyncServiceImpl::BookmarkNodeRemoved(
+    bookmarks::BookmarkModel* model,
+    const bookmarks::BookmarkNode* parent,
+    int old_index,
+    const bookmarks::BookmarkNode* node,
+    const std::set<GURL>& no_longer_bookmarked) {
+
```

Comment:
not yet implemeted completely, but looks straightforward taking to attention all changes.